### PR TITLE
Create a side-panel component with customizable actions

### PR DESCRIPTION
**Pull request by deioo**
## Responsive Side-panel Component styles

**Preview - Large**
screen shot 2015-03-31 at 17 42 38

**Preview - Medium**
screen shot 2015-03-31 at 17 42 55

**Preview - Mobile**
screen shot 2015-03-31 at 17 43 07

Introduce sass maps to allow having a custom hash for custom side-panel actions:

Single custom settings variable:
```scss
$side-panel-custom-actions: (
  'navigation': (
    previous: angle-left,
    next: angle-right
  ),
  'project': (
    featured: star,
    add: plus,
    like: thumbs-o-up
 )
);
```

Custom actions Markup:
```html
<ul class="navigation-actions">
  <li><button class="previous"></button></li>
  <li><button class="next"></button></li>
</ul>

<ul class="project-actions">
  <li><button class="featured"></button></li>
  <li><button class="add"></button></li>
  <li><button class="like"></button></li>
</ul>
```

Output:
image

### CODE REVIEW

File: components/_side-panel.scss [hunk 1]
```diff
@@ -0,0 +1,154 @@
+$side-panel-actions-button-size: 35px;
+$side-panel-actions-height: rem-calc(60);
+$side-panel-small-screen-size: 45em;
+$side-panel-medium-screen-size: 75em;
+$side-panel-large-screen-size: 90em;
+$side-panel-xlarge-screen-size: 120em;
+$side-panel-custom-actions: (
```
Comment (matteodepalo) [inline components/_side-panel.scss]: I think renaming `custom-actions` to actions should not be confusing since there are no other types of actions.

**Comment by matteodepalo** [comment]
Apart from one comment the code looks amaziiiiiiing! :fireworks:

File: components/_side-panel.scss (hunk 1 above)
Comment (deioo) [inline components/_side-panel.scss]: Done :)

**Comment by matteodepalo** [comment]
:+1:

**Pull request merged**